server/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -390,7 +390,7 @@ public boolean shouldCache(Query query) {
         persistMetadata(path, indexSettings, shardRouting, null, logger);
         this.useRetentionLeasesInPeerRecovery = replicationTracker.hasAllPeerRecoveryRetentionLeases();
         this.refreshPendingLocationListener = new RefreshPendingLocationListener();
-        this.isDataStreamIndex = mapperService == null ? false : mapperService.mappingLookup().isDataStreamTimestampFieldEnabled();
+        this.isTimeseriesIndex = mapperService == null ? false : mapperService.mappingLookup().hasTimestampField();
```

Comment:
Is this needed ? This information can be accessed through the mapping directly, no ? Can we keep the `isDataStreamIndex` since this information is useful outside of this PR ?

Reply:
Addressed in f852ec983a3b3dc670b308f6215f612424b27e21